addons/auth_signup/controllers/main.py

```diff
@@ -148,10 +148,12 @@ def get_auth_signup_qcontext(self):
 
     def _prepare_signup_values(self, qcontext):
         values = { key: qcontext.get(key) for key in ('login', 'name', 'password') }
+        password = values.get('password')
         if not values:
             raise UserError(_("The form was not properly filled in."))
-        if values.get('password') != qcontext.get('confirm_password'):
+        if password != qcontext.get('confirm_password'):
             raise UserError(_("Passwords do not match; please retype them."))
+        request.env['res.users']._check_password_policy([password])
```

Comment:
I checked the UI flow related to this code, and it seems to have changed slightly. Could you verify it?Do we need this code as per the new flow change? Can you please check?